### PR TITLE
Fix: How to check the image extension from Uri.

### PR DESCRIPTION
**Pull request by rustbasic**
Fix: How to check the image extension from Uri.

* Related #5324
* Related #5341

It addresses most of what was discussed in #5341, but I'm not sure if it addresses everything.

### CODE REVIEW

**Comment by xangelix** [comment]
Thank you for the PR and apologies again that my previous changes have caused issues for you, though I have concerns that this PR would still fail in some web URI scenarios.

Also, if this were to be the direction we go, you'd almost certainly want to do something like this to avoid having to keep extensions in sync with the image crate manually :)

```rs
    ImageFormat::all()
        .filter(ImageFormat::reading_enabled)
        .flat_map(ImageFormat::extensions_str)
        .any(|format_ext| uri.contains(*format_ext))
```

**Comment by rustbasic** [comment]
> Thank you for the PR to resolve an issue I introduced and apologies that my previous changes have caused issues for you, though I have concerns that this PR would still fail in some web URI scenarios.
> 
> Also, if this were to be the direction we go, you'd almost certainly want to do something like this to avoid having to keep extensions in sync with the image crate manually :)
> 
> ```rust
>     ImageFormat::all()
>         .filter(ImageFormat::reading_enabled)
>         .flat_map(ImageFormat::extensions_str)
>         .any(|format_ext| uri.contains(*format_ext))
> ```

I have modified this PR as you said.
If you want to make a PR, I will close this PR once you submit it.

File: crates/egui_extras/src/loaders/image_loader.rs [hunk 1]
```diff
@@ -19,16 +19,11 @@ impl ImageCrateLoader {
 }
 
 fn is_supported_uri(uri: &str) -> bool {
-    let Some(ext) = Path::new(uri).extension().and_then(|ext| ext.to_str()) else {
-        // `true` because if there's no extension, assume that we support it
-        return true;
-    };
-
     // Uses only the enabled image crate features
```
Comment (rustbasic) [inline crates/egui_extras/src/loaders/image_loader.rs:22]: In most cases it's fine, but I'm not sure if it's a good idea to remove this part.

**Comment by emilk** [comment]
Closed by https://github.com/emilk/egui/pull/5431

**Pull request closed without merging**